pipeline_dp/budget_accounting.py

```diff
@@ -1,20 +1,91 @@
 """Privacy budget accounting for DP pipelines."""
 
 
+class Budget:
+  """Manages the budget allocated for an operation."""
+
+  def __init__(self, weight, use_eps, use_delta, eps=None, delta=None):
+    """Constructs a Budget.
+
+    The values for eps and delta will be computed only after the pipeline was
+    constructed and the method compute_budgets was called.
+
+    Args:
+      weight: The weight is used to get more accurate results.
+      use_eps: A boolean that is False when the operation doesn't need epsilon.
+      use_delta: A boolean that is False when the operation doesn't need delta.
+    """
+    self._eps = eps
+    self._delta = delta
+    self.weight = weight
+    self.use_eps = use_eps
+    self.use_delta = use_delta
+
+  @property
+  def eps(self):
+    """eps: Optional; Parameter of (eps, delta)-differential privacy."""
```

Comment:
Could you please remove eps: Optional?

Please add: Raise an error if the budget is not calculated yet. 


(the same with delta).